torch/nn/modules/rnn.py

```diff
@@ -55,7 +55,14 @@ def reset_parameters(self):
         for weight in self.parameters():
             weight.data.uniform_(-stdv, stdv)
 
-    def forward(self, input, hx):
+    def forward(self, input, hx=None):
+        if (hx == None):
+            batch_sz = input.size()[0] if self.batch_first else input.size()[1]
+            hx = torch.autograd.Variable(torch.Tensor(self.num_layers, batch_sz,
+                                                   self.input_size).zero_())
+            if (self.mode == 'LSTM'):
+                hx = (torch.autograd.Variable(hx.data),
+                      torch.autograd.Variable(hx.data))
```

Comment:
Use the same Variable twice, it's not supported to have many `Variable`s referencing the same tensor.
`hx = (hx, hx)`